pkg/controller/jobframework/integrationmanager.go

```diff
@@ -198,6 +199,20 @@ func (m *integrationManager) getList() []string {
 	return ret
 }
 
+func (m *integrationManager) isKnownOwner(ownerRef *metav1.OwnerReference) bool {
+	for _, cbs := range m.integrations {
+		if cbs.matchingOwnerReference(ownerRef) {
+			return true
+		}
+	}
+	for _, jt := range m.externalIntegrations {
+		if ownerReferenceMatchingGVK(ownerRef, jt.GetObjectKind().GroupVersionKind()) {
+			return true
+		}
+	}
+	return ownerRef.Kind == "ReplicaSet" && ownerRef.APIVersion == "apps/v1"
```

Comment:
The first two blocks of code make sense to me.

The replicaSet does not. I'm sure this is correct but could you add a comment here explaining why?

Reply:
Sure, we need to be able to walk up the ownerhip chain for Pod -> ReplicaSet -> Deployment. 

However, ReplicaSet is "technical detail" rather than an integration a user would like to enable or see in the config.

Reply:
/hold
We have had one e2e test for this scenario, but we are missing unit test, I will add.

Reply:
/unhold
added under "Pod -> ReplicaSet -> Deployment (queue-name) => Deployment". I tested it fails without this line.

Reply:
Makes sense.  Thanks for adding the extra test case.

Reply:
will do soon

Reply:
Done, ptal